### DISCUSSION TITLE
How to correctly set latex-babel for pdf output

### DISCUSSION THREAD

**Discussion by produnis** [quarto-dev/quarto-cli#2368] [Q&A] [answered]
Hi there,
I am creating a thesis-template using quarto-book. While everything else is working as intended, I am struggeling to set up latex-babel for german language support for PDF output. I want "List of Figures" and "List of Tables" to be in german language, and I need support for the german ß Symbol.

A "working" -not working- example can be found here: https://git.zilium.de/produnis/quarto-thesis-hsnr

`git clone git@git.zilium.de:produnis/quarto-thesis-hsnr.git`

In `_quarto.yml` I specified
```
book:
  lang: de-DE

format:
  pdf:
    documentclass: scrbook
    classoption:
      - ngerman
    include-in-header: "latex-header.tex" 
```
See full yml here: https://git.zilium.de/produnis/quarto-thesis-hsnr/-/blob/main/_quarto.yml

In [`latex-header.tex`](https://git.zilium.de/produnis/quarto-thesis-hsnr/-/blob/main/latex-header.tex) I added

`\usepackage[ngerman]{babel}`

and I even tried to force it the hard way  with
```
\renewcommand{\listfigurename}{Abbildungsverzeichnis}
\renewcommand{\listtablename}{Tabellenverzeichnis}
\renewcaptionname{ngerman}{\listfigurename}{Abbildungsverzeichnis}
\renewcaptionname{ngerman}{\listtablename}{Tabellenverzeichnis}

```

but in my PDF output file, there is still "*List of Figures*" and "*List of Tables*" instead of "*Abbildungsverzeichnis*"

![](https://i.imgur.com/grmPji5.png)

And the ß-Symbol is rendered as *SS*.

![no ß but SS](https://i.imgur.com/I2nuUgW.png)

![no ß but SS](https://i.imgur.com/RUhWLyD.png)

These steps all work with my other pure-latex-only-projects, so I am out of ideas how to make it work with quarto.

Does anyone have a working example on how to switch latex output to `[ngerman]{babel}` ?

**Comment by produnis** [comment, marked as answer]
I solved the "ß"-problem by removing `\usepackage[T1]{fontenc}`.....

...but still no clue how to change "List of FIgures" and "List of Tables"....

**Reply by cderv** [reply]
I believe this kind of switch to use babel are controlled by the `lang` variable, which will control in Pandoc the babel usage in LaTeX, and that we use for translation in Quarto too (https://quarto.org/docs/authoring/language.html)

This previous discussion is also related I believe https://github.com/quarto-dev/quarto-cli/issues/757 (but issue has been fixed). 

List of Figure and List of Table translation are supported by Quarto I believe. `lang: de-DE` should have picked up the `de` translation file 🤔 
https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/language/_language-de.yml

Those values will be used to inject the correct TeX to renew `listfigurename` as you did. 

Can you add `keep-tex` and look at the generated TeX file maybe to see how this looks ? 

I'll try to reproduce this

**Reply by produnis** [reply]
This is the tex-output. As you can see at the last lines of this code, my commands are overwritten by quarto...
```
% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
\PassOptionsToPackage{dvipsnames,svgnames,x11names}{xcolor}
%
\documentclass[
  12pt,
  a4paperpaper,
  headsepline,
  ngerman,
  oneside,
  smallheadings,
  openany,
  liststotoc,
  bibtotoc,
  DIV=calc]{scrbook}

\usepackage{amsmath,amssymb}
\usepackage{lmodern}
\usepackage{setspace}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
  \setmainfont[]{Times New Roman}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\setlength{\emergencystretch}{3em} % prevent overfull lines
\setcounter{secnumdepth}{5}
% Make \paragraph and \subparagraph free-standing
\ifx\paragraph\undefined\else
  \let\oldparagraph\paragraph
  \renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
  \let\oldsubparagraph\subparagraph
  \renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi


\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}\usepackage{longtable,booktabs,array}
\usepackage{calc} % for calculating minipage widths
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newlength{\csllabelwidth}
\setlength{\csllabelwidth}{3em}
\newlength{\cslentryspacingunit} % times entry-spacing
\setlength{\cslentryspacingunit}{\parskip}
\newenvironment{CSLReferences}[2] % #1 hanging-ident, #2 entry spacing
 {% don't indent paragraphs
  \setlength{\parindent}{0pt}
  % turn on hanging indent if param 1 is 1
  \ifodd #1
  \let\oldpar\par
  \def\par{\hangindent=\cslhangindent\oldpar}
  \fi
  % set entry spacing
  \setlength{\parskip}{#2\cslentryspacingunit}
 }%
 {}
\usepackage{calc}
\newcommand{\CSLBlock}[1]{#1\hfill\break}
\newcommand{\CSLLeftMargin}[1]{\parbox[t]{\csllabelwidth}{#1}}
\newcommand{\CSLRightInline}[1]{\parbox[t]{\linewidth - \csllabelwidth}{#1}\break}
\newcommand{\CSLIndent}[1]{\hspace{\cslhangindent}#1}

\usepackage[utf8]{inputenc}
\usepackage{fontspec}           % System-Fontfamilien xelatex
%\usepackage[ngerman]{babel}     % deutsche Silbentrennung
\usepackage[numbered]{bookmark}
\usepackage{graphicx}
\usepackage{tikz}
\usepackage{eurosym}
\usepackage{wrapfig}
\usepackage{tabularx}
\usepackage{colortbl}
\usepackage{multirow}
\usepackage{amsmath}%
\usepackage[normalem]{ulem}
\usepackage{hhline}
\usepackage{rotating}
\usepackage{xcolor}
\usepackage{caption}
\setlength\parindent{0pt}      % Keine Texteinrückung am Absatzanfang


 \renewcommand{\listfigurename}{Abbildungsverzeichnis}
 \renewcommand{\listtablename}{Tabellenverzeichnis}
 \renewcaptionname{ngerman}{\listfigurename}{Abbildungsverzeichnis}
 \renewcaptionname{ngerman}{\listtablename}{Tabellenverzeichnis}
 \addto\captionsngerman{%
   \renewcommand{\listfigurename}{Abbildungsverzeichnis}
   \renewcommand{\listtablename}{Tabellenverzeichnis}
 }
\makeatletter
\makeatother
\makeatletter
\@ifpackageloaded{bookmark}{}{\usepackage{bookmark}}
\makeatother
\makeatletter
\@ifpackageloaded{caption}{}{\usepackage{caption}}
\AtBeginDocument{%
\ifdefined\contentsname
  \renewcommand*\contentsname{Table of contents}
\else
  \newcommand\contentsname{Table of contents}
\fi
\ifdefined\listfigurename
  \renewcommand*\listfigurename{List of Figures}
\else
  \newcommand\listfigurename{List of Figures}
\fi
\ifdefined\listtablename
  \renewcommand*\listtablename{List of Tables}
\else
  \newcommand\listtablename{List of Tables}

( ... )
```

**Reply by cderv** [reply]
ok thanks. The command inserted by Quarto should use the translation file if you correctly set the `lang` option, as explained in https://quarto.org/docs/authoring/language.html#lang-option
so `lang: de` should load the `_language-de.yaml` that has the translation 
https://github.com/quarto-dev/quarto-cli/blob/5bfc2ec863ffa3c3e8f4f4a40352bae19d8c372e/src/resources/language/_language-de.yml#L66-L68

I see now that you put it under `book` - Do you use VSCODE or RSTUDIO IDE ? Validation should help you know if this is valid. I believe it is not an option under book, but should be under the format. 
* See PDF reference: https://quarto.org/docs/reference/formats/pdf.html#language
* Book reference has not `lang` (https://quarto.org/docs/reference/projects/books.html)

I think where you put it, it did not get picked up, even by Pandoc - hence the need to add yourself `\usepackage[ngerman]{babel}` I believe.

You can also provide custom translation in the YAML, so you could change the Quarto default without all the translation
https://quarto.org/docs/authoring/language.html#alternate-language

**Reply by produnis** [reply]
thank you so much!!!!!!!!!!!!
removing `lang: de-DE` from `book` and adding `lang: de` to format-pdf solved my problem!
```
format:
  pdf:
    lang: de
```

with this, there is no need for my crossref language specifications either, everything is working as intended.

THANK YOU!!!!

**Reply by cderv** [reply]
Awesome ! Let's mark this as answer. Sorry to not have seen the yaml misplacement right away.

**Reply by fuhrmanator** [reply]
Hello - sounds like this is related to https://github.com/quarto-dev/quarto-cli/issues/3059#issuecomment-1292667398 (missing locale)?
